test/e2e/common.go

```diff
@@ -133,3 +138,13 @@ func getCloudProviderConfig(cluster, vmType string) (string, error) {
 	}
 	return base64.StdEncoding.EncodeToString(b), err
 }
+
+func redactLogs() {
```

Comment:
side note: do you think it'd be easy to do the same with collecting workload cluster logs at some point?

Reply:
`redactLogs()` should already be redacting workload cluster logs as long as they are stored in `${ARTIFACTS}`.

Reply:
sorry I meant collect logs for e2e, since right now we're only running `log-dump.sh` on conformance clusters

Reply:
I think the CAPI test framework is already collecting a lot of relevant logs for us, similar to `log-dump.sh`, though different folder structure (e.g. https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/746/pull-cluster-api-provider-azure-capi-e2e/1278793437226733568/artifacts/)

Reply:
AFAIK it's collecting bootstrap cluster logs (CAPI/CAPZ managers, etc.) but not logs from the workload cluster VMs (eg. cloud-init logs)

Reply:
maybe that's something we should add to the framework itself

Reply:
Similar to the other issue in this PR, I can enable VM log collection in a follow-up PR. And yea this is definitely somewhere we could contribute back to CAPI.